docs(link): add `custom` prop to samples

ashleahhill opened the pull request:
The `tag` prop of `<router-link>` has been removed in v4.0.0+

https://github.com/vuejs/rfcs/blob/master/active-rfcs/0021-router-link-scoped-slot.md

The `custom` prop is functional, however it is not mentioned anywhere in the docs.

<!--
Please make sure to read the Pull Request Guidelines:
https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#pull-request-guidelines
-->

posva commented on docs/api/README.md:
````diff
@@ -113,17 +115,36 @@ If you add a `target="_blank"` to your `a` element, you must omit the `@click="n
   ```
 
-### tag
+### tag (3.1.0 < 4.0.0)
````
can you remove all changes after this one (including it). The docs of v4 are at https://next.router.vuejs.org/

ashleahhill replied:
Well, I feel silly now. I didn't find that link anywhere yesterday :-P

ashleahhill committed:
docs(router-link) Add `custom` prop

To use a cusom tag for `router-link`, v4.0.0+ requires the use of the `custom` prop.

The `custom` prop is functional, however it is not in the code examples.

posva reviewed (approved):
Thanks!
